src/utility.hpp

```diff
@@ -821,6 +835,40 @@ string file_base_name(const string& filename);
 /// Only works for files readable by the current user.
 bool file_exists(const string& filename);
 
+/// Determine if a file can be written to.
+/// Only works for files writable by the current user.
+/// Meant to be used to validate output file names.
+/// Will actually open the file, so it will overwrite the current contents.
+bool file_can_be_written(const string& filename);
+
+/// Check if a file exists and return its name (if so) or error.
+/// Uses file_exists() and is intended to be called when parsing arguments.
+string error_if_file_does_not_exist(const string& context, const string& filename);
+
+/// Error if a file looks like it's gzipped.
+void error_if_file_is_gzipped(const string& context, const string& filename);
+
+/// Check if a file can be written to and return its name (if so) or error.
+/// Uses file_can_be_written() and is intended to be called when parsing arguments.
+string error_if_file_cannot_be_written(const string& context, const string& filename);
+
+/// A special parser for thread count which errors if non-positive
+/// If max_threads is non-zero, it will also decrease to that maximum.
+/// If max_threads is zero, max is omp_get_max_threads()
+int parse_thread_count(const string& context, const string& arg, int max_threads = 0);
```

Comment:
Rather than enforcing that `omp_set_num_threads()` and `parse_thread_count()` are used together with a checker script, why not just make one function that sets the thread count from a string and always call that?